site/docs/hosting/fly.md

```diff
@@ -0,0 +1,426 @@
+# Hosting: Fly <Badge text="Deno" /><Badge text="Node.js" type="warning"/>
+
+This guide tells you about the ways you can host your grammY bots on [Fly](https://fly.io), either using Deno or Node.js.
+
+## Preparing Your Code
+
+You can run your bot either on [webhooks](../guide/deployment-types.md) or [long polling](../guide/deployment-types.md).
+
+### Webhooks
+
+> Remember that you should not call `bot.start()` in your code when using webhooks.
+
+You can use the example code provided in our [Deno Deploy](./deno-deploy.md#preparing-your-code) (Deno) and [Heroku](./heroku.md#express-and-its-middleware) (Node.js) hosting guide.
+In this section, we literally transcribe the text from the both docs.
+
+1. Make sure that you have a file which exports your `Bot` object, so that you can import it later to run it.
+2. Create a file named `app.ts` or `app.js`, or actually any name you like (but you should be remembering and using this as the main file to deploy), with the following content:
+
+<CodeGroup>
+<CodeGroupItem title="Deno" active>
+
+```ts{6,11}
+import { serve } from "https://deno.land/std/http/server.ts";
+import { webhookCallback } from "https://deno.land/x/grammy/mod.ts";
+// You might modify this to the correct way to import your `Bot` object.
+import { bot } from "./bot.ts";
+
+const port = 8000;
+const handleUpdate = webhookCallback(bot, "std/http");
+
+await serve(async (req) => {
+  const url = new URL(req.url);
+  if (req.method == "POST" && url.pathname.slice(1) == bot.token) {
+    try {
+      return await handleUpdate(req);
+    } catch (err) {
+      console.error(err);
+    }
+  }
+  return new Response();
+}, { port });
+```
+
+</CodeGroupItem>
+<CodeGroupItem title="Node.js" active>
+
+```ts{6,10}
+import express from "express";
+import { webhookCallback } from "grammy";
+// You might modify this to the correct way to import your `Bot` object.
+import { bot } from "./bot";
+
+const port = 8000;
+const app = express();
+
+app.use(express.json());
+app.use(`/${bot.token}`, webhookCallback(bot, "express"));
+app.use((_req, res) => res.status(200).send());
+
+app.listen(port, () => console.log(`listening on port ${port}`));
+```
+
+</CodeGroupItem>
+</CodeGroup>
+
+We advise you to have your handler on some secret path rather than the root (`/`).
+Here, we are using the bot token (`/<bot token>`).
+
+### Long Polling
+
+Create a file named `app.ts` or `app.js`, or actually any name you like (but you should be remembering and using this as the main file to deploy), with the following content:
+
+<CodeGroup>
+<CodeGroupItem title="Deno" active>
+
+```ts
+import { Bot } from "https://deno.land/x/grammy/mod.ts";
+
+const bot = new Bot(""); // <-- insert the bot token here
```

Comment:
Perhaps this guide is late enough in the development process to educate people about environment variables, and link them to https://fly.io/docs/rails/the-basics/configuration/?